index.js

```diff
@@ -13,25 +14,45 @@ function HyperdriveSwarm (archive, opts) {
   this.uploading = !(opts.upload === false)
   this.downloading = !(opts.download === false)
   this.live = !(opts.live === false)
+  this.authorize = !(opts.authorize === false)
+
+  var isHyperdbInstance = !!(archive.get && archive.put && archive.replicate && archive.authorize)
```

Comment:
Something like this will be needed in quite a few places, we also stumbled upon this need when making hyper-graph-db compatible with subhyperdb. What about adding a property `isHyperdb` to the hyperdb object to make this simpler and more error-proof?

Reply:
+1 @mafintosh

Reply:
@Frando do you have any time to open a PR on hyperdb for this?

Reply:
out of curiosity, why not use `this instanceOf HyperDB`?

Reply:
@micahscopes ah that wouldn't work because this module doesn't `require` hyperdb or hypercore

Reply:
@karissa is spot on, good old duck typing. 🦆